### PR TITLE
feat(theme-common): code block MagicComments support for Lua/Haskell -- and WebAssembly ;;

### LINKED ISSUE #8859: Add support for Lua comment style for MagicComments
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

The current magic comment syntax for highlighting is great but it doesn't have support for Lua's comment style.
Lua uses `--` for short single line comments. Right now I resort to using to using the C/Python comment styles as a workaround but this makes copy/pasting/testing the code snippets a pain since I need to keep removing these invalid lines.

### Reproducible demo

_No response_

### Steps to reproduce

- Use lua language in CodeBlock
- use magic comments to highlight regions of code

### Expected behavior

Expected `-- highlight-start` and `-- highlight-end` to work in lua but it does not. 

### Actual behavior

Does not work

### Your environment

_No response_

### Self-service

- [ ] I'd be willing to fix this bug myself.

### PR DESCRIPTION
**Pull request by tarunrajput**
Add support for Lua, haskell, SQL and wasm comment style for MagicComments. Related #8859 

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #8859) and the maintainers have approved on my working plan.

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/
https://deploy-preview-8870--docusaurus-2.netlify.app/tests/pages/code-block-tests

## Related issues/PRs

https://github.com/facebook/docusaurus/issues/8859

### CODE REVIEW

File: packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts [hunk 1]
```diff
@@ -83,6 +84,10 @@ function getAllMagicCommentDirectiveStyles(
       // Text uses HTML, front matter uses bash
       return getCommentPattern(['html', 'jsx', 'bash'], magicCommentDirectives);
 
+    case 'lua':
+    case 'haskell':
+      return getCommentPattern(['lua'], magicCommentDirectives);
+
     default:
       // All comment types
       return getCommentPattern(
```
Comment (Josh-Cena) [inline packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts:93]: You should remove the `lua` pattern by default. There are too many false-positives

Comment (tarunrajput) [reply packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts]: You're right. I have updated the PR

**Comment by tarunrajput** [comment]
> Thanks for your PR
> 
> How can I review it?
> 
> Can you provide me a test link on the preview?
> 
> I suggest creating 3 new code blocks (lua,haskell,wasm) on `/docusaurus/website/_dogfooding/_pages tests/code-block-tests.mdx` so that we can see the output on the preview

@slorber,   I have included the dogfooding tests, as well as added support for SQL, as its single-line comment syntax is similar to that of Lua and Haskell.

https://deploy-preview-8870--docusaurus-2.netlify.app/tests/pages/code-block-tests

File: website/_dogfooding/_pages tests/code-block-tests.mdx [hunk 2]
```diff
@@ -267,3 +267,48 @@ export default function MyReactPage() {
   );
 }
 ```
+
```
Comment (slorber) [inline website/_dogfooding/_pages tests/code-block-tests.mdx:270]:
```suggestion
## Magic comments tests

```

File: website/_dogfooding/_pages tests/code-block-tests.mdx [hunk 3]
```diff
@@ -267,3 +267,49 @@ export default function MyReactPage() {
   );
 }
 ```
+## Magic comments tests
```
Comment (slorber) [inline website/_dogfooding/_pages tests/code-block-tests.mdx:270]:
```suggestion

## Magic comments tests
```

**Review by slorber — APPROVED** [review]
Thanks, LGTM 👍

**Pull request merged**